catalog-spec/catalog-best-practices.md

```diff
@@ -0,0 +1,119 @@
+# STAC Catalog Best Practices
+
+This document makes a number of recommendations for creating real world SpatioTemporal Asset Catalogs. None of them 
+are required to meet the core specification, but following these practices will make life easier for client tooling
+and for users. They come about from practical experience of implementors, and introduce a bit more 'constraint' for
+those who are creating new catalogs or new tools to work with STAC. 
+
+In time some of these may evolve to become part of the core specification, but while the current goal of the core is to remain 
+quite flexible and simple to meet a wide variety of use cases.
+
+[Work In Progress]
+
+## Catalog Layout
+
+Creating a catalog involves a number of decisions as to what folder structure to use to represent sub-catalogs, items
+and assets, and how to name them. The specification leaves this totally open, and you can link things as you want. We
+encourage people to explore new structures. But the following are what a number of implementors ended up doing. Following
+these recommendations makes for more legible catalogs.
+
+1. Root documents (catalogs / collections) should be at the root of a directory tree containing the static catalog.
+2. Catalogs should be named `catalog.json` (cf. `index.html`).
+3. Collections that are distinct from catalogs should be named `collection.json`.
+4. Items should be named `<id>.json`
+5. Sub-catalogs should be stored in subdirectories of their parent (and only 1 subdirectory deeper than a document's parent) (e.g. `.../sample/sub1/catalog.json`).
+6. Items should be stored in subdirectories of their parent catalog (e.g. `.../sample/sub1/3425/3425.json`). 
```

Comment:
I'd change the example to remove the directory, ```../sample/sub1/3425.json```, just to make clear that it is not required for an item to have a unique subdir with the same name, see https://github.com/radiantearth/stac-spec/issues/408#issuecomment-468242880. I assume that recommendation (7) below was created based on https://github.com/radiantearth/stac-spec/issues/408#issuecomment-468369974, and the idea there is that if the number of itens in a subdir is 'reasonable' then it is OK to have more than one item in a single subdir.

Reply:
Agreed, this needs more clarification, though `.../sample/sub1/3425.json` differs from `.../sample/sub1/3425/3425.json` in the depth.

Probably the former if assets are limited or external, the latter if there are many assets.

Reply:
Cool, will just remove the example.